examples/chef/create_docker.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import optparse
+import tarfile
+import shutil
+import docker
+
+client = docker.from_env()
+
+_CREATE_DOCKER_SCRIPT_PATH = os.path.dirname(__file__)
+_SUPPORTED_PLATFORM = 'linux_x86'
+
+parser = optparse.OptionParser()
+parser.add_option('-c', '--commit_sha', type='string',
+                  action='store',
+                  dest='commit_sha')
+parser.add_option('-s', '--short_sha', type='string',
+                  action='store',
+                  dest='short_sha')
+parser.add_option('-r', '--revision_id', type='string',
+                  action='store',
+                  dest='revision_id')
+parser.add_option('-b', '--build_id', type='string',
+                  action='store',
+                  dest='build_id')
+parser.add_option('-i', '--image_name', type='string',
+                  action='store',
+                  dest='image_name')
+parser.add_option('-t', '--tar_path', type='string',
+                  action='store',
+                  dest='tar_path')
+
+options, args = parser.parse_args(sys.argv[1:])
+
+# Check that all options are set
+for option, value in options.__dict__.items():
```

Comment:
use argparse and `required` instead. I think `__dict__` is internal due to the underscores.

Reply:
Followed this suggestion. Thanks.